src/test/java/com/programmers/springbootbasic/domain/voucher/application/VoucherServiceTest.java

```diff
@@ -0,0 +1,167 @@
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.when;
+
+import com.programmers.springbootbasic.domain.TestVoucherIdGenerator;
+import com.programmers.springbootbasic.domain.voucher.domain.VoucherIdGenerator;
+import com.programmers.springbootbasic.domain.voucher.domain.VoucherRepository;
+import com.programmers.springbootbasic.domain.voucher.domain.VoucherType.VoucherTypeEnum;
+import com.programmers.springbootbasic.domain.voucher.domain.entity.Voucher;
+import com.programmers.springbootbasic.domain.voucher.presentation.dto.CreateVoucherRequest;
+import com.programmers.springbootbasic.domain.voucher.presentation.dto.UpdateVoucherRequest;
+import com.programmers.springbootbasic.domain.voucher.presentation.dto.VoucherResponse;
+import com.programmers.springbootbasic.exception.exceptionClass.VoucherException;
+import java.util.List;
+import java.util.Optional;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.springframework.test.util.ReflectionTestUtils;
+
+class VoucherServiceTest {
+
+    @InjectMocks
+    private VoucherService voucherService;
+    @Mock
+    private VoucherRepository voucherRepository;
+    private VoucherIdGenerator idGenerator;
+
+    @BeforeEach
+    public void setUp() {
+        MockitoAnnotations.openMocks(this);
+        this.idGenerator = new TestVoucherIdGenerator();
+        ReflectionTestUtils.setField(voucherService, "idGenerator", idGenerator);
+    }
+
+    @DisplayName("유효한 정보의 바우처를 생성 할 수 있다.")
+    @Test
+    void success_create() {
+        // given
+        CreateVoucherRequest request = CreateVoucherRequest.of(VoucherTypeEnum.FIXED.name(), 10);
+        when(voucherRepository.save(any(Voucher.class))).thenReturn(
+            new Voucher(idGenerator.generate(), request.getVoucherType(),
+                request.getBenefitValue()));
+
+        // when
+        var result = voucherService.create(request);
+
+        // then
+        assertEquals(result, idGenerator.generate());
+    }
+
+    @DisplayName("모든 바우처를 조회 할 수 있다.")
+    @Test
+    void success_findAll() {
+        // given
+        when(voucherRepository.findAll()).thenReturn(
+            List.of(
+                new Voucher(idGenerator.generate(), VoucherTypeEnum.FIXED.getVoucherType(10), 10),
+                new Voucher(idGenerator.generate(), VoucherTypeEnum.PERCENT.getVoucherType(10), 10)
+            )
+        );
+
+        // when
+        var result = voucherService.findAll();
+
+        // then
+        assertEquals(2, result.size());
+        for (Object item : result) {
+            assertTrue(item instanceof VoucherResponse);
+        }
+    }
```

Comment:
```suggestion
    private static Stream<Arguments> voucherTypes() {
        return Stream.of(
            Arguments.of(new TestVoucherIdGenerator().generate(), VoucherTypeEnum.FIXED.getVoucherType(10), 10),
            Arguments.of(new TestVoucherIdGenerator().generate(), VoucherTypeEnum.PERCENT.getVoucherType(10), 10)
        );
    }

    @MethodSource("voucherTypes")
    @ParameterizedTest(name = "바우처 타입: {0}, 혜택 값: {1}")
    void testCase1(UUID id, VoucherType voucherType, int benefitValue) {
        // given
        var voucher = new Voucher(id, voucherType, benefitValue);
        when(voucherRepository.findById(id)).thenReturn(Optional.of(voucher));

        // when
        var result = voucherService.findById(id);

        // then
        assertThat(result.getVoucherType()).isEqualTo(voucherType.getVoucherTypeName());
        assertThat(result.getBenefitValue()).isEqualTo(benefitValue);
    }
```

전체 조회와 검증하려고 하는 부분을 MethodSource를 통해 fixture를 정의해놓고, 테스트 메서드에는 필요한 명령어만으로 가독성을 높일 수 있을 것 같습니다.